src/http/matcher.cc

```diff
@@ -65,7 +66,12 @@ size_t param_matcher::match(const sstring& url, size_t ind, parameters& param) {
         }
         return sstring::npos;
     }
-    param.set(_name, url.substr(ind, last - ind));
+    sstring path;
+    if (http::internal::url_decode(url.substr(ind, last - ind), path)) {
```

Comment:
Also note that with the path decoding already done in `parse_query_param`, the argument `url` here is already decoded, so there is no need to decode the path component a second time. (Let me know if I am missing something.)